lib/tasks/seek_upgrades.rake

```diff
@@ -84,6 +85,20 @@ namespace :seek do
     puts " ... finished copying project ids of #{decoupled.to_s} extracted samples"
   end
 
+  task(link_sample_datafile_attributes: [:environment]) do
+    puts '... updating sample_resource_links for samples with data_file attributes...'
+    samples_updated = 0
+    disable_authorization_checks do
+      df_attrs = SampleAttribute.joins(:sample_attribute_type).where('sample_attribute_types.base_type' => Seek::Samples::BaseType::SEEK_DATA_FILE).pluck(:id)
+      samples = Sample.joins(sample_type: :sample_attributes).where('sample_attributes.id' => df_attrs)
+      samples.each do |sample|
+        sample.send(:update_sample_resource_links)
+        samples_updated += 1
+      end
+    end
```

Comment:
This looks more complicated than it should need to be. If necessary, it should be put in a class with tests.
Also looks like in terms or repeatability, each time it is run it will replace the resource_links with new ones, so should skip if they've already been created

Reply:
I've added an if to only update the links when they are empty in [1fffe06](https://github.com/seek4science/seek/commit/1fffe069b9d89245c01726582d5f708b0d914a85).
I am failing to see what is so complicated... I am just getting a list of samples that have df attributes and updating their links... I thought this would be much more efficient than looping through all the sample types and then their samples...